[Bug]: Alter alias not working after restarting docker compose, collection failed  to load

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 2.0.2
- Deployment mode(standalone or cluster): standalone
- SDK version(e.g. pymilvus v2.0.0rc2): pymilvus 2.0.2
- OS(Ubuntu or CentOS): Ubuntu 20.04
- CPU/Memory: Intel Core i7-11700k / 16gb
- GPU: -
- Others: -
```


### Current Behavior

After I shutdown the service using docker compose down, and reinitialize the service again using the docker compose up. Failed to alter alias and some collection ( not all ) failed to load. 

As a context as milvus doesn't have list alias API yet, I try to embed the collection name into it's description, so to know which alias map to whic collection, I just describe the alias collection and read the description to find the collection it's mapped to. Something like this ( the alias is `propertyUnits` and the literal collection name is `propertyUnits__v0_1`

```
<Collection>:
-------------
<name>: propertyUnits
<partitions>: [{"name": "_default", "collection_name": "propertyUnits", "description": ""}]
<description>: propertyUnits__v0_1
...
}
``` 

I found that after I downed and restarted the docker compose, alter alias not working directly. The alias only changed if I restarted the service again and the cycle continue. 

In addition I also encounter a collection load error ( only 1 apart from 4 collection I have ):

```
2022-06-07 07:14:21,155 : ERROR : RPC error: [wait_for_loading_collection], <BaseException: (code=1, message=collection 433716601692094465 has not been loaded to memory or load failed)>, <Time:{'RPC start': '2022-06-07 07:14:20.651431', 'RPC error': '2022-06-07 07:14:21.155698'}>
2022-06-07 07:14:21,155 : ERROR : RPC error: [load_collection], <BaseException: (code=1, message=collection 433716601692094465 has not been loaded to memory or load failed)>, <Time:{'RPC start': '2022-06-07 07:14:20.635769', 'RPC error': '2022-06-07 07:14:21.155832'}>
*** pymilvus.client.exceptions.BaseException: <BaseException: (code=1, message=collection 433716601692094465 has not been loaded to memory or load failed)>
``` 
The order of sequence I do :

1. docker compose up
2. create collections `propertyUnits__v0_1`
3. create collections `propertyUnits__v0_2`
4. insert data to both collections
5. create alias `propertyUnits` and assign to `propertyUnits__v0_1`
6. describe the `propertyUnits` alias ( working properly, mapped to `propertyUnits__v0_1` )
7. alter alias `propertyUnits` to `propertyUnits__v0_2`
8. describe the `propertyUnits` alias ( working properly, mapped to `propertyUnits__v0_2` )
9. docker compose down
10. docker compose up
11. alter alias `propertyUnits` to `propertyUnits__v0_1`
12. describe the `propertyUnits` alias ( not working, still mapped to `propertyUnits__v0_2` )
13. docker compose down
14. docker compose up
15. describe the `propertyUnits` alias ( now mapped to `propertyUnits__v0_1` )
16. alter alias `propertyUnits` to `propertyUnits__v0_2`
17. describe the `propertyUnits` alias ( not working, still mapped to `propertyUnits__v0_1` )
18. docker compose down
19. docker compose up
20. describe the `propertyUnits` alias ( now mapped to `propertyUnits__v0_2`)


### Expected Behavior

- Alter alias working immediately even after service restart
- No collection load error

### Steps To Reproduce

```markdown
The order of sequence I do :

1. docker compose up
2. create collections `propertyUnits__v0_1`
3. create collections `propertyUnits__v0_2`
4. insert data to both collections
5. create alias `propertyUnits` and assign to `propertyUnits__v0_1`
6. describe the `propertyUnits` alias ( working properly, mapped to `propertyUnits__v0_1` )
7. alter alias `propertyUnits` to `propertyUnits__v0_2`
8. describe the `propertyUnits` alias ( working properly, mapped to `propertyUnits__v0_2` )
9. docker compose down
10. docker compose up
11. alter alias `propertyUnits` to `propertyUnits__v0_1`
12. describe the `propertyUnits` alias ( not working, still mapped to `propertyUnits__v0_2` )
13. docker compose down
14. docker compose up
15. describe the `propertyUnits` alias ( now mapped to `propertyUnits__v0_1` )
16. alter alias `propertyUnits` to `propertyUnits__v0_2`
17. describe the `propertyUnits` alias ( not working, still mapped to `propertyUnits__v0_1` )
18. docker compose down
19. docker compose up
20. describe the `propertyUnits` alias ( now mapped to `propertyUnits__v0_2`)
```


### Milvus Log

No error on alter alias

Error on collection load : 

```
2022-06-07 07:14:21,155 : ERROR : RPC error: [wait_for_loading_collection], <BaseException: (code=1, message=collection 433716601692094465 has not been loaded to memory or load failed)>, <Time:{'RPC start': '2022-06-07 07:14:20.651431', 'RPC error': '2022-06-07 07:14:21.155698'}>
2022-06-07 07:14:21,155 : ERROR : RPC error: [load_collection], <BaseException: (code=1, message=collection 433716601692094465 has not been loaded to memory or load failed)>, <Time:{'RPC start': '2022-06-07 07:14:20.635769', 'RPC error': '2022-06-07 07:14:21.155832'}>
*** pymilvus.client.exceptions.BaseException: <BaseException: (code=1, message=collection 433716601692094465 has not been loaded to memory or load failed)>
```

### Anything else?

_No response_

I've also found that after restarting docker compose, the alias appeared as collection themselves ( they appear when I try to list collection using `utility.list_collections`  ).

I also try dropping the collection which is linked to the alias ( without dropping the alias) and recreate the collection and reassign the alias, it gives me this error 

```
milvus-standalone | [2022/06/08 14:36:19.247 +00:00] [ERROR] [root_coord.go:2024] ["CreateAlias failed"] [role=rootcoord] [alias=developerProjects_house] ["collection name"=developerProjects_house__v0_1] [msgID=433768914909660395] [error="meta table add alias failed, error = collection alias collides with existing collection name. collection = developerProjects_house, alias = developerProjects_house"] [stack="github.com/milvus-io/milvus/internal/rootcoord.(*Core).CreateAlias\n\t/go/src/github.com/milvus-io/milvus/internal/rootcoord/root_coord.go:2024\ngithub.com/milvus-io/milvus/internal/distributed/rootcoord.(*Server).CreateAlias\n\t/go/src/github.com/milvus-io/milvus/internal/distributed/rootcoord/service.go:82\ngithub.com/milvus-io/milvus/internal/proto/rootcoordpb._RootCoord_CreateAlias_Handler.func1\n\t/go/src/github.com/milvus-io/milvus/internal/proto/rootcoordpb/root_coord.pb.go:933\ngithub.com/grpc-ecosystem/go-grpc-middleware/tracing/opentracing.UnaryServerInterceptor.func1\n\t/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/tracing/opentracing/server_interceptors.go:38\ngithub.com/milvus-io/milvus/internal/proto/rootcoordpb._RootCoord_CreateAlias_Handler\n\t/go/src/github.com/milvus-io/milvus/internal/proto/rootcoordpb/root_coord.pb.go:935\ngoogle.golang.org/grpc.(*Server).processUnaryRPC\n\t/go/pkg/mod/google.golang.org/grpc@v1.38.0/server.go:1286\ngoogle.golang.org/grpc.(*Server).handleStream\n\t/go/pkg/mod/google.golang.org/grpc@v1.38.0/server.go:1609\ngoogle.golang.org/grpc.(*Server).serveStreams.func1.2\n\t/go/pkg/mod/google.golang.org/grpc@v1.38.0/server.go:934"]
```

full log in here 
[milvus_2.log](https://github.com/milvus-io/milvus/files/8862693/milvus_2.log)
@soothing-rain

--- a/internal/rootcoord/meta_table.go
+++ b/internal/rootcoord/meta_table.go
@@ -151,6 +151,19 @@ func (mt *MetaTable) reloadFromKV() error {
 		mt.proxyID2Meta[proxyMeta.ID] = proxyMeta
 	}
 
+	_, values, err = mt.snapshot.LoadWithPrefix(CollectionAliasMetaPrefix, 0)
+	if err != nil {
+		return err
+	}
+	for _, value := range values {
+		aliasInfo := pb.CollectionInfo{}
+		err = proto.Unmarshal([]byte(value), &aliasInfo)
+		if err != nil {
+			return fmt.Errorf("rootcoord Unmarshal pb.AliasInfo err:%w", err)
+		}
+		mt.collAlias2ID[aliasInfo.Schema.Name] = aliasInfo.ID
+	}
+
 	_, values, err = mt.snapshot.LoadWithPrefix(CollectionMetaPrefix, 0)
 	if err != nil {
 		return err
@@ -161,6 +174,9 @@ func (mt *MetaTable) reloadFromKV() error {
 		err = proto.Unmarshal([]byte(value), &collInfo)
 		if err != nil {
 			return fmt.Errorf("rootcoord Unmarshal pb.CollectionInfo err:%w", err)
+		}
+		if _, ok := mt.collAlias2ID[collInfo.Schema.Name]; ok {
+			continue
 		}
 		mt.collID2Meta[collInfo.ID] = collInfo
 		mt.collName2ID[collInfo.Schema.Name] = collInfo.ID
@@ -217,19 +233,6 @@ func (mt *MetaTable) reloadFromKV() error {
 			return fmt.Errorf("rootcoord Unmarshal pb.IndexInfo err:%w", err)
 		}
 		mt.indexID2Meta[meta.IndexID] = meta
-	}
-
-	_, values, err = mt.snapshot.LoadWithPrefix(CollectionAliasMetaPrefix, 0)
-	if err != nil {
-		return err
-	}
-	for _, value := range values {
-		aliasInfo := pb.CollectionInfo{}
-		err = proto.Unmarshal([]byte(value), &aliasInfo)
-		if err != nil {
-			return fmt.Errorf("rootcoord Unmarshal pb.AliasInfo err:%w", err)
-		}
-		mt.collAlias2ID[aliasInfo.Schema.Name] = aliasInfo.ID
 	}
 
 	log.Debug("reload meta table from KV successfully")

--- a/internal/rootcoord/meta_table_test.go
+++ b/internal/rootcoord/meta_table_test.go
@@ -345,6 +345,18 @@ func TestMetaTable(t *testing.T) {
 	})
 
 	wg.Add(1)
+	t.Run("not load alias when load collection meta", func(t *testing.T) {
+		defer wg.Done()
+		ts := ftso()
+		err = mt.AddAlias(aliasName1, collName, ts)
+		assert.Nil(t, err)
+		err = mt.reloadFromKV()
+		assert.Nil(t, err)
+		_, ok := mt.collName2ID[aliasName1]
+		assert.False(t, ok)
+	})
+
+	wg.Add(1)
 	t.Run("add partition", func(t *testing.T) {
 		defer wg.Done()
 		ts := ftso()

